Validation/HGCalValidation/python/hgcalPlots.py

```diff
@@ -0,0 +1,1660 @@
+import os
+import copy
+import collections
+
+import six
+import ROOT
+ROOT.gROOT.SetBatch(True)
+ROOT.PyConfig.IgnoreCommandLineOptions = True
+
+from Validation.RecoTrack.plotting.plotting import Plot, PlotGroup, PlotFolder, Plotter, PlotOnSideGroup
+from Validation.RecoTrack.plotting.html import PlotPurpose
+import Validation.RecoTrack.plotting.plotting as plotting
+import Validation.RecoTrack.plotting.validation as validation
+import Validation.RecoTrack.plotting.html as html
+
+#To be able to spot any issues both in -z and +z a layer id was introduced 
+#that spans from 0 to 103 for hgcal_v9 geometry. The mapping for hgcal_v9 is: 
+#-z: 0->51
+#+z: 52->103
+layerscheme = { 'lastLayerEEzm': 28, 'lastLayerFHzm': 40, 'maxlayerzm': 52, 'lastLayerEEzp': 80, 'lastLayerFHzp': 92, 'maxlayerzp': 104 }
```

Comment:
This is marginally better, but really this information should be derived directly from the geometry in C++.

Reply:
Do you mean something like taking it straight from here for example: 

https://github.com/cms-sw/cmssw/blob/master/RecoLocalCalo/HGCalRecAlgos/interface/HGCalImagingAlgo.h#L168-L174

or do you mean reading it from the xml files of the geometry?

Reply:
No, don't use hardcoded values from another header file.

The geometry info can be accessed through the RecHit tools, which get the information from the geometry classes, e.g.:
https://github.com/cms-sw/cmssw/blob/41a74ae88fdc8c2fa69447b66df4916ab23c953b/RecoLocalCalo/HGCalRecAlgos/interface/RecHitTools.h#L58-L59

This kind of access is almost guaranteed to avoid failure in case of changes in the geometry, while hardcoded values are almost guaranteed to fail eventually.

Reply:
Hi @kpedro88 , 

I see that lastLayerFH() in RecHitTools gives the last layer. Could it be possible to have something like:  

unsigned int lastLayerEE() const {return eelastlayer;} 
unsigned int lastLayerFH() const {return fhlayer ;} 
unsigned int lastLayerBH() const {return bhlastlayer;} 

I mean, I am missing the last layer of FH which is needed.

Reply:
@apsallid in the new geometry, the last layer of "FH" is the same as the last layer of "BH", because the silicon/scintillator boundary is variable.

@bsunanda would it be possible to add some functions like "lastLayerFine" and "lastLayerCoarse" that denote the change in longitudinal segmentation?

Reply:
Anticipating these really helpful functions, I am pushing a solution to avoid hardcoded values. 
Since this package is going to take heavy development in the weeks to come, do you think we could go on with a solution like that? I mean, 
use the hardcoded values in hgcalPlots.py but the moment the new functions will become available remove it since the code is 
already there?

Reply:
@apsallid this is acceptable, thanks

Reply:
I opened issue #26225 to track this